### ISSUE TITLE
Missing referential integrity when insert into child table (prisma-client-js) without the existence of parent row 

### ISSUE THREAD

**Issue by ManolisPap** [prisma/prisma#13242]
### Bug description

I use PlaneScale as a database. I have create a parent-child relation and i used prisma referential Integity preview feature.
But does not seems to work. I can insert a child row even if parent row that does not exists.

In my case i have a Order table (Parent) and an OrderStatusHistory (Child table). 
OrderStatusHistory has as primary key the order_id, but i can insert a new OrderStatusHistory row with an orderId i.e., 1000 even if there is no order into Orders table with that id.

### How to reproduce

I have create a public project that you can clone if you like:
https://github.com/ManolisPap/prisma-playground

1. Create an empty database with PlaneScale
2. create the .env file with an envirome variable DATABASE_URL='...'
3. create the prisma.schema file (with the attached code)
4. npx prisma db push
5. create the index.ts file (with the attached code)
6. run the index.ts file `npx ts-node index.ts` 
7. The typescript code will run successfully and we will create a new row into OrderStatusHistory table even the Order Table is empty (prisma does not check the referential integrity)

Or just follows the step from prisma.io:
[https://www.prisma.io/docs/getting-started/setup-prisma/start-from-scratch/relational-databases-typescript-postgres](Initial Setup)

### Expected behavior

Primsa returns an error that could not insert an orderStatusHistory row because there is missing an order on Orders table with that orderId.

### Prisma information

My database url: 
DATABASE_URL='mysql://user:pscale_pw@ip/database_name**?connection_limit=5&sslaccept=strict'**

Or clone the project: https://github.com/ManolisPap/prisma-playground

**prisma.schema**

```
datasource db {
  provider             = "mysql"
  url                  = env("DATABASE_URL")
  referentialIntegrity = "prisma"
}

generator client {
  provider        = "prisma-client-js"
  previewFeatures = ["referentialIntegrity"]
}

enum OrderStatus {
  NEW
  PICKED
  STORED
  DELIVERED
}

model Order {
  orderId          Int                  @id @default(autoincrement())
  statusMilestones OrderStatusHistory[]

  @@index([orderId])
}

model OrderStatusHistory {
  orderStatusHistoryId Int         @id @default(autoincrement())
  orderId              Int
  status               OrderStatus
  createdAt            DateTime    @default(now())
  order                Order       @relation(fields: [orderId], references: [orderId], onUpdate: Restrict, onDelete: Cascade)

  @@index([orderId])
}
``` 

**Dummy Typescript usage**

``` 
console.log("Prisma");
import { PrismaClient } from "@prisma/client";

const prisma = new PrismaClient({
  log: ["query", "info", "warn", "error"],
  errorFormat: "pretty",
});

async function main() {
  const result = await prisma.orderStatusHistory.create({
    data: {
      status: "NEW",
      orderId: 1000,
    },
  });

  console.log(result);
}

main()
  .catch((e) => {
    throw e;
  })
  .finally(async () => {
    await prisma.$disconnect();
  });
``` 

### Environment & setup

- OS: Windows
- Database: Planetscale
- Node.js version: v16.13.0

### Prisma Version

```
prisma: "^3.13.0"
```

**Comment by Weakky** [comment]
Internal note: I suspect this to be related to https://www.notion.so/prismaio/MongoDB-extended-referential-integrity-5861c4818f104080b41fb7d04f851afd and to not be a bug

**Comment by ManolisPap** [comment]
@Weakky I am not sure that i can access/join the notion.so that you send. Can you give a different link or past here the text?
Thanks in advance!

**Comment by dpetrick** [comment]
When writing key values directly instead of using `connect` or other operations, we're not guaranteeing referential integrity - this is our so-called "unchecked" operation path. There are valid use-cases for databases without foreign keys to write incomplete data across a system and being eventually consistent, so we don't want to prevent these use cases by being overly zealous.

Tl;dr: Use `connect` instead of writing `orderId` directly.
